templates/plugin/tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "esModuleInterop": true,
+        "allowSyntheticDefaultImports": true,
+        "target": "es6",
+        "moduleResolution": "node",
+        "sourceMap": true,
+        "outDir": "dist",
+        "baseUrl": ".",
```

Comment:
Since we are targeting Node.js and not an AMD I would completely drop both, `baseUrl` and `paths` since they could create "bizzare" issues, hard to debug, and should increase the rate of issues on fastify repos that are related to TS. You can find more info here: https://www.typescriptlang.org/docs/handbook/module-resolution.html#base-url